packages/taiko-client-rs/crates/driver/src/metrics.rs

```diff
@@ -0,0 +1,48 @@
+//! Metrics exposed by the driver runtime.
+
+use metrics::{Unit, describe_gauge};
+
+/// Metric namespace for the driver.
+pub struct DriverMetrics;
+
+impl DriverMetrics {
+    /// Gauge tracking the latest L1 origin block ID seen on the execution engine.
+    pub const BEACON_HEAD_BLOCK_ID: &'static str = "driver_beacon_head_block_id";
+    /// Gauge tracking the next proposal ID reported by the inbox core state.
+    pub const NEXT_PROPOSAL_ID: &'static str = "driver_next_proposal_id";
+    /// Gauge tracking the last finalized proposal ID.
+    pub const LAST_FINALIZED_PROPOSAL_ID: &'static str = "driver_last_finalized_proposal_id";
+    /// Gauge tracking the highest proposal ID observed from events.
+    pub const LAST_SEEN_PROPOSAL_ID: &'static str = "driver_last_seen_proposal_id";
+    /// Gauge tracking the number of pending transitions waiting for finalization.
+    pub const TRANSITION_QUEUE_DEPTH: &'static str = "driver_transition_queue_depth";
+
```

Comment:
actually all of them are unused expect the first one `BEACON_HEAD_BLOCK_ID`

Reply:
Yes, the `metrics` part is still (mostly) unimplemented, i think it's ok to keep it there for now, will update all metrics stuff in the next PR.